gnoi/os/server.go

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2020 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package os
+
+import (
+	"context"
+
+	"github.com/google/gnxi/gnoi/os/pb"
+	"google.golang.org/grpc"
+)
+
+// Server is an OS Management service.
+type Server struct {
+	pb.OSServer
+	manager *Manager
+}
+
+// NewServer returns an OS Management service.
+func NewServer(factoryVersion string) *Server {
+	return &Server{manager: NewManager(factoryVersion)}
```

Comment:
I think it is worth using a OSSettings struct defined in manager.go to pass:
factoryVersion string
installedVersions []string

just like reset does.